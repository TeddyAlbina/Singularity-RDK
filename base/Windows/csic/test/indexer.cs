using System;
class T {
	int x;
	public int this[int a, T b] {
		get { return a+x; }
		set { x=b.x=value; }
	}
	public T(int x) { this.x = x; }
	public void print() { Console.WriteLine("{0}", x); }
	static public void Main() {
		T a = new T(1), b = new T(10);
		a.print(); b.print();
		Console.WriteLine("{0}", a[2,b]);
		a[2,b] = 20;
		a.print(); b.print();
		Console.WriteLine("{0}", a[2,b]++);
		b.print();
		a[2,b] += 5;
		a.print(); b.print();
	}
}
