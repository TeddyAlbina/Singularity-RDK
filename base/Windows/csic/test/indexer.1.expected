1
10
3
20
20
22
23
30
30
